ccdproc/image_collection.py

```diff
@@ -652,10 +663,17 @@ def _generator(self, return_type,
 
             file_name = path.basename(full_path)
 
+            if 'unit' not in ccd_kwargs:
```

Comment:
I don't see how `test_headers` can access the `fits_ccddata_reader`. I will have a look.